sqlglot/parser.py

```diff
@@ -2177,27 +2179,33 @@ def _parse_table_sample(self) -> t.Optional[exp.Expression]:
         size = None
         seed = None
 
-        self._match_l_paren()
+        kind = "TABLESAMPLE" if self._prev.token_type == TokenType.TABLE_SAMPLE else "USING SAMPLE"
```

Comment:
Maybe we should be using a boolean flag here instead..

Reply:
not sure